File: html_tags/open_close_tag.py
# 열고 닫는 태그
def open_close_tag(tag_name,  childrens=[], types=[], values=[]):
    # 태그 열기
    tag_start = f"<{tag_name}"

    # type 추가
    def type_plus(tag_start, types,values):
        # tag 지정
        tag = tag_start
        # 만약에 values가 list면서 빈배열이 아닐때
        if type(values) and values != []:
            # type들 모두 추가
            for index, type_name in enumerate(types):
                # type 및 values의 인덱스에 해당하는 값들 추가
                tag += f" {type_name}={values[index]}"
        # values가 list면서 빈배열일때
        if type(values) and values == []:
            # type들 모두 추가
            for type_name in types:
                # tag에 type 모두 대입
                tag += f" {type_name}"

        # type들 모두 추가한 후,
        # 닫아버리기(시작 태그 기준)
        tag += ">"
        return tag

    # 시작 태그 지정
    open_tag = type_plus(tag_start, types,values)

    # 태그 내의 result 지정
    # children 값들 모두 추가
    result_init = ""

    # 배열 childrens을 분해하여
    # result에 추가
    def children_plus(result_init, childrens):
        # open_tag에 children 배열들 모두 추가
        # children으로 나눔
        for children in childrens:
            # children 추가
            result_init += children
        # children 값들 모두 return
        return result_init

    # 결과들 대입
    result = children_plus(result_init, childrens)

    # close_tag 지정
    close_tag = f"</{tag_name}>"

    # 총 태그 지정
    total_tag = open_tag + result + close_tag

    # 출력
    return total_tag

File: html_tags/openCloseTag.js
import openTag from "./tag.js";

// 열고 닫는 태그를 제작하는 함수
// 클로저 형식 함수
// 첫번째 파라마터 : tagName (태그이름)
// 두번째 파라미터 : types (속성명, 배열형식)
// 세번째 파라미터 : values (속성값, 배열형식)
// 네번째 파라미터 ㅣ chidren(태그 내부에 들어가는 내용, 배열 형식)


// types, values, children 모두 기본값 빈 배열
const openCloseTag = (tagName, types = [], values = [], children=[])=> {

    // 시작 태그
    // openTag 함수 import
    const tagStart = openTag(tagName, types, values);

    // 닫는 태그
    const tagEnd = `</${tagName}>`

    // 들어갈 내용
    let content = ''
    // 클로저 함수를 통해
    // content에 들어갈 내용 추가
    const childrenPlus = (children) => {

        // children 분해 
        children.forEach(element => {
            // content에 element 추가
            content += element;
        });
    }

    // content에 내용 추가
    childrenPlus(children);

    // 최종 값 생성
    const result = tagStart + content + tagEnd;

    // result 반환
    return result;

}